packages/main/src/MenuItem.ts

```diff
@@ -201,6 +207,54 @@ class MenuItem extends ListItem implements IMenuItem {
 	@i18n("@ui5/webcomponents")
 	static i18nBundle: I18nBundle;
 
+	_itemNavigation: ItemNavigation;
+	_lastFocusedItemIndex: number | null;
+
+	constructor() {
+		super();
+
+		this._itemNavigation = new ItemNavigation(this, {
+			getItemsCallback: () => this._navigableItems,
+		});
+		this._lastFocusedItemIndex = null;
+	}
+
+	get _navigableItems() {
+		let navigableItems: Array<HTMLElement> = [];
+
+		if (!this.hasEndContent) {
+			return [];
+		}
+
+		navigableItems.push(...this.endContent);
+
+		return navigableItems;
+	}
+
+	_itemKeyDown(e: KeyboardEvent) {
+		if (isLeft(e)) {
+			this._handleNextOrPreviousItem(e);
+		} else if (isRight(e)) {
+			this._handleNextOrPreviousItem(e, true);
+		}
+	}
+
+	_handleNextOrPreviousItem(e: KeyboardEvent, isNext?: boolean) {
+		const target = e.target as MenuItem | HTMLElement;
+
+		let updatedTarget = target;
```

Comment:
Do we need the `updatedTarget` variable. Can't we use the `target` constat directly?

Reply:
Done